temporalio/converter.py

```diff
@@ -112,8 +133,8 @@ async def decode(
         raise NotImplementedError
 
 
-class CompositeDataConverter(DataConverter):
-    """Composite data converter that delegates to a list of payload converters.
+class CompositePayloadConverter(PayloadConverter):
```

Comment:
Why extend PayloadConverter and not implement its methods?

Reply:
I do implement the two abstract methods here: `to_payloads` and `from_payloads`. If I didn't, this would have been a runtime error.

Reply:
Oh I missed this last night, isn't it simpler to require implementing `to_payload` and `from_payload`? I was expecting that.

Reply:
The multi-payload form seem to have the most use cases (granted we sometimes call with one payload). `PayloadConverter`, `PayloadCodec`, and `DataConverter` all operate on collections of payloads/values. Only `EncodingPayloadConverter` operates on single payloads/values since it is metadata specific.

Also, technically we support combining or expanding multiple to/from one. Encode documents that it must return at least one and no more than given. Decode documents that it must return at least one.